app/services/bolognese/writers/hyku_addons_work_form_fields_writer.rb

```diff
@@ -4,37 +4,38 @@
 # Add new fields here that should be available to prefill the work forms.
 # You must also add the logic to the PrefillWorkFormViaDOI JS class.
 #
-# Raw output from this class can be accessed via - Copy the raw source, not the HTML output:
-# http://YOUR_TENANT.lvh.me:3000/doi/autofill?curation_concern=generic_work&doi=YOUR_DOI
+# The processed JSON can be access via a JSON request:
+# # http://YOUR_TENANT.lvh.me:3000/doi/autofill.xml?curation_concern=generic_work&doi=YOUR_DOI
 #
-# E.g
-# http://repo.lvh.me:3000/doi/autofill?curation_concern=generic_work&doi=10.7554/elife.63646
+# To create fixtures for specs, the the unprocessed XML can be accessed via an XML request.
+# NOTE: Copy the raw source, not the HTML output:
+# http://YOUR_TENANT.lvh.me:3000/doi/autofill.xml?curation_concern=generic_work&doi=YOUR_DOI
+
+# rubocop:disable Metrics/ModuleLength
 module Bolognese
   module Writers
     module HykuAddonsWorkFormFieldsWriter
+      include HykuAddons::WorkFormNameable
+
       DATE_FORMAT = "%Y-%-m-%-d"
       DOI_REGEX = /10.\d{4,9}\/[-._;()\/:A-Z0-9]+/
       ROR_QUERY_URL = "https://api.ror.org/organizations?query="
 
-      def hyku_addons_work_form_fields
-        {
-          "identifier" => Array(identifiers).select { |id| id["identifierType"] != "DOI" }.pluck("identifier"),
-          "doi" => Array(doi),
-          "title" => titles&.pluck("title"),
-          "creator" => write_involved("creators"),
-          "contributor" => write_involved("contributors"),
-          "funder" => write_funders,
-          "publisher" => Array(publisher),
-          "date_published" => write_date_published,
-          "abstract" => write_descriptions,
-          "keyword" => subjects&.pluck("subject"),
-          "official_link" => Array(url),
-          "language" => Array(language)
-        }.compact.reject { |_key, value| value.blank? }
+      def hyku_addons_work_form_fields(curation_concern: "generic_work")
+        @curation_concern = curation_concern
+
+        # Work through each of the work types fields to create the data hash
+        form_data = work_type_terms.each_with_object({}) do |term, data|
+          method_name = "write_#{term}"
+
+          data[term.to_s] = respond_to?(method_name, true) ? send(method_name) : nil
```

Comment:
I think you can substitue the full sentence for a single `try`:

data[term.to_s] = try(method_name)

Reply:
It seems that `try` uses `public_send` which wont work for protected methods: 

https://github.com/rails/rails/blob/main/activesupport/lib/active_support/core_ext/object/try.rb#L7